### PR TITLE
pass context to trace functions

### PR DESCRIPTION
**Pull request by hmoragrega**
enables passing the context to the trace functions

NOTE: this is a breaking change, the cloud kit is updated here
https://github.com/netdata/cloud-kit/pull/190

### CODE REVIEW

File: paho/client.go [hunk 1]
```diff
@@ -377,7 +377,7 @@ func (c *Client) writer() {
 		}
 		_, err := w.WriteTo(c.Conn)
 		if err != nil {
```
Comment (nicolasparada) [inline paho/client.go:380]: Perhaps do we want to skip `context canceled` errors?
Or at least don't log them with error level.

Comment (hmoragrega) [reply paho/client.go:380]: done 🙋   I've changed the level to warning in such case

**Review by nicolasparada — APPROVED** [review]
👍

**Review by lvrach — APPROVED** [review]
ctx everywhere

**Pull request merged**